src/cleo/commands/completions_command.py

```diff
@@ -249,34 +249,62 @@ def sanitize(s: str) -> str:
         # Commands + options
         cmds = []
         cmds_opts = []
-        cmds_names = []
+        namespaces = set()
         for cmd in sorted(self.application.all().values(), key=lambda c: c.name or ""):
             if cmd.hidden or not cmd.enabled or not cmd.name:
                 continue
-            command_name = shell_quote(cmd.name) if " " in cmd.name else cmd.name
-            cmds.append(
-                f"complete -c {script_name} -f -n '__fish{function}_no_subcommand' "
-                f"-a {command_name} -d '{sanitize(cmd.description)}'"
-            )
+            cmd_path = cmd.name.split(" ")
+            namespace = cmd_path[0]
+            cmd_name = cmd_path[-1] if " " in cmd.name else cmd.name
+
+            # We either have a command like `poetry add` or a nested (namespaced)
+            # command like `poetry cache clear`.
+            if len(cmd_path) == 1:
+                cmds.append(
+                    f"complete -c {script_name} -f -n '__fish{function}_no_subcommand' "
+                    f"-a {cmd_name} -d '{sanitize(cmd.description)}'"
+                )
+                condition = f"__fish_seen_subcommand_from {cmd_name}"
+            else:
```

Comment:
I don't think this is quite enough because we have commands like `poetry self show plugins` which are nested deeper than 2 levels, so we would need to use some recurrence here to make completions for each namespace level (not sure how it works on the `fish` side). Not sure if it's something we want to go that deep with changes with the current state of the code.

Reply:
Ah forgot about that one, but yea I'm not sure it's worth the effort since it's the only command that deep and takes no flags